sklearn/model_selection/_validation.py

```diff
@@ -754,53 +779,76 @@ def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params,
-                'technique resulting in properly '
-                'stratified folds')
-            warnings.warn('Number of classes in training fold ({}) does '
-                          'not match total number of classes ({}). '
-                          'Results may not be appropriate for your use case. '
-                          '{}'.format(len(estimator.classes_),
-                                      n_classes, recommendation),
-                          RuntimeWarning)
-            if method == 'decision_function':
-                if (predictions.ndim == 2 and
-                        predictions.shape[1] != len(estimator.classes_)):
-                    # This handles the case when the shape of predictions
-                    # does not match the number of classes used to train
-                    # it with. This case is found when sklearn.svm.SVC is
-                    # set to `decision_function_shape='ovo'`.
-                    raise ValueError('Output shape {} of {} does not match '
-                                     'number of classes ({}) in fold. '
-                                     'Irregular decision_function outputs '
-                                     'are not currently supported by '
-                                     'cross_val_predict'.format(
-                                        predictions.shape, method,
-                                        len(estimator.classes_),
-                                        recommendation))
-                if len(estimator.classes_) <= 2:
-                    # In this special case, `predictions` contains a 1D array.
-                    raise ValueError('Only {} class/es in training fold, this '
-                                     'is not supported for decision_function '
-                                     'with imbalanced folds. {}'.format(
-                                        len(estimator.classes_),
-                                        recommendation))
-
-            float_min = np.finfo(predictions.dtype).min
-            default_values = {'decision_function': float_min,
-                              'predict_log_proba': float_min,
-                              'predict_proba': 0}
-            predictions_for_all_classes = np.full((_num_samples(predictions),
-                                                   n_classes),
-                                                  default_values[method])
-            predictions_for_all_classes[:, estimator.classes_] = predictions
-            predictions = predictions_for_all_classes
+        if isinstance(predictions, list):
+            predictions = [_enforce_prediction_order(
+                estimator.classes_[i_label], predictions[i_label],
+                n_classes=len(set(y[:, i_label])), method=method)
+                for i_label in range(len(predictions))]
+        else:
+            # A 2D y array should be a binary label indicator matrix
```

Comment:
it can't be otherwise multioutput here?

Reply:
I'm pretty sure it can't, no. If the predictions are a single array and not a list, and there's a 2D array of targets, then the targets are binary. Multiclass multi-target problems produce lists of arrays as output.

Reply:
Right, in this context. Good.